stetho/src/main/java/com/facebook/stetho/inspector/elements/ObjectDescriptor.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+package com.facebook.stetho.inspector.elements;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class ObjectDescriptor extends Descriptor {
+  private final Map<Class<?>, String> mClassToFriendlyNameCache =
+      Collections.synchronizedMap(new HashMap<Class<?>, String>());
+
+  @Override
+  public void hook(Object element) {
+  }
+
+  @Override
+  public void unhook(Object element) {
+  }
+
+  @Override
+  public NodeType getNodeType(Object element) {
+    return NodeType.ELEMENT_NODE;
+  }
+
+  @Override
+  public String getNodeName(Object element) {
+    return getClassFriendlyName(element.getClass());
+  }
+
+  private String getClassFriendlyName(Class<?> elementClass) {
+    if (mClassToFriendlyNameCache.containsKey(elementClass)) {
+      return mClassToFriendlyNameCache.get(elementClass);
+    }
+
+    String className = elementClass.getName();
+
+    String friendlyClassName =
+        removePrefix(className, "android.app.",
```

Comment:
This is at the wrong layer of inheritance.  Only `View` subclasses can be in `android.{view,widget}.*`, and only `Activity` can be in `android.app.*`.  The substitutions should be happening in `ViewDescriptor` and `ActivityDescriptor` as a result.